Docs/StyleGuide/Style-Guide.md

```diff
@@ -0,0 +1,280 @@
+This style guide provides coding conventions for the Dafny code.
+
+*This documentation is still in progress. Please feel free to add any more suggestions if any.*
+
+## Naming Convention
+Any **variables** are named with `camelCase`.
+```dafny
+var minValue := 1;
+var cipherMessage := "Hello World";
+```
+
+Any **lemmas**, **predicates**, **functions**, **methods**, **classes**, and **modules** are named with `PascalCase`.
+```dafny
+method FindIndex(arr: seq<int>, k: int)
+    ...
+```
+
+Any static or global **constants** are named with `UPPERCASE_WITH_UNDERSCORES`.
+```dafny
+static const MONTHS_IN_A_YEAR := 12;
+```
+
```

Comment:
What about datatypes, datatype constructors and the fields of a datatype? Maybe there are other things I'm not thinking of right now?

Reply:
For now, I will write `datatype`s and `newtype`s should follow `PascalCase`, but I'm not sure if this is the best practice. What do other people think?